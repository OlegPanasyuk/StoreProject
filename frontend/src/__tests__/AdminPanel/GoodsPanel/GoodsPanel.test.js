import React from 'react';
import GoodsPanelConnected, { GoodsPanel } from '../../../AdminPanel/GoodsPanel/GoodsPanel';
import { shallow, configure, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    filters: {  
        limit: 10,
        count: 0,
        activePage: 1,
        priceMore: 0,
        priceLess: 0,
        nameSearch: '',
        orderPrice: null,
        id_catalogue: -1
    },
    goodsFilter: ()=>{},
    closeEditGoodsItem: ()=>{},
    deleteItem: {},
    permissionToDelete: ()=>{},
    permissionToDeleteClose: ()=>{},
    editItem: {},
    goodsToShow: []
}) {
    let component = shallow(<GoodsPanel {...props} />);
    return { component, props };
}

describe('GoodsPanel --- Snapshot', () => {
    it('Component must be rendered', () => {
        let { component } = setupComponent();
        expect(GoodsPanel.propTypes.goodsToShow).toBe(PropTypes.array);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});


describe('GoodsPanel ', () => {
    let initialState = {
        adminPanel_goodsPanel: {
            goodsToShow: [],
            filters: {
                limit: 10,
                count: 0,
                activePage: 1,
                priceMore: 0,
                priceLess: 0,
                nameSearch: '',
                orderPrice: null,
                id_catalogue: -1
            },
            editItem: {
                show: false,
                idgoods: null,
                name: null,
                description: null,
                price: 0,
                catalogue_id_catalogue: null
            },
            deleteItem: {
                id: null,
                show: false
            }

        }
    };

    const mockStore = configureStore();
    let store, wrapper;
    
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(<Provider store={store}><GoodsPanelConnected /></Provider>);
    });
    it('+++ render the connected(SMART) component',() => {
        expect(wrapper.find(GoodsPanelConnected).length).toEqual(1);
    });
});

