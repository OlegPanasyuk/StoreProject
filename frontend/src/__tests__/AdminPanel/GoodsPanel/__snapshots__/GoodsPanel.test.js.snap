// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoodsPanel --- Snapshot Component must be rendered 1`] = `
<Fragment>
  <div />
  <div />
  <div />
  <Container
    as="div"
    fluid={false}
  >
    <Bootstrap(Row)>
      <Col
        as="div"
        className="text-center"
        xs={12}
      >
        <ListOfPages
          activePage={1}
          count={0}
          limit={10}
          openPage={[Function]}
        />
      </Col>
    </Bootstrap(Row)>
    <Bootstrap(Row)>
      <Col
        as="div"
        className="text-center d-flex justify-content-center"
        xs={12}
      >
        Price:
        <a
          href="#up"
          onClick={[Function]}
        >
          Up
        </a>
        <a
          href="#down"
          onClick={[Function]}
        >
          Down
        </a>
      </Col>
    </Bootstrap(Row)>
    <Bootstrap(Row)>
      <Col
        as="div"
        xs={2}
      >
        <Button
          active={false}
          disabled={false}
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          Add goods
        </Button>
      </Col>
      <Col
        as="div"
        className="d-flex flex-column justify-content-between"
        xs={8}
      />
      <Col
        as="div"
        xs={2}
      >
        <FilterGoodsPanel
          countFiltered={[Function]}
          openPage={[Function]}
          prepSearchRow={[Function]}
          updateState={[Function]}
        />
      </Col>
    </Bootstrap(Row)>
    <Bootstrap(Row)>
      <Col
        as="div"
        className="text-center"
        xs={12}
      >
        <ListOfPages
          activePage={1}
          count={0}
          limit={10}
          openPage={[Function]}
        />
      </Col>
    </Bootstrap(Row)>
  </Container>
</Fragment>
`;
